code/modules/mob/living/basic/farm_animals/mushroom.dm

```diff
@@ -0,0 +1,185 @@
+	///Where we store our cap icons so we dont generate them constantly to update our icon
+	var/static/mutable_appearance/cap_living
+	///Where we store our cap icons so we dont generate them constantly to update our icon
+	var/static/mutable_appearance/cap_dead
+	///So you can't repeatedly revive it during a fight
+	COOLDOWN_DECLARE(recovery_cooldown)
+
+/mob/living/basic/mushroom/Initialize(mapload)
+	melee_damage_lower = rand(4, 6)
+	melee_damage_upper = rand(11,21)
+	maxHealth = rand(50,70)
+	cap_living = cap_living || mutable_appearance(icon, "mushroom_cap")
+	cap_dead = cap_dead || mutable_appearance(icon, "mushroom_cap_dead")
+	cap_color = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
+	UpdateMushroomCap()
+	health = maxHealth
+	. = ..()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_WALKING_MUSHROOM, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+/datum/ai_controller/basic_controller/mushroom
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/mushroom(),
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree/mushroom,
+		/datum/ai_planning_subtree/find_and_hunt_target/mushroom_food,
+	)
+
+
+/datum/targetting_datum/basic/mushroom
+	stat_attack = DEAD
+
+///we only attacked another mushrooms
+/datum/targetting_datum/basic/mushroom/faction_check(mob/living/living_mob, mob/living/the_target)
+	return !living_mob.faction_check_mob(the_target, exact_match = check_factions_exactly)
+
+
+/datum/ai_planning_subtree/basic_melee_attack_subtree/mushroom
+	melee_attack_behavior = /datum/ai_behavior/basic_melee_attack/mushroom
+
+/datum/ai_behavior/basic_melee_attack/mushroom
+	action_cooldown = 2 SECONDS
+
+/datum/ai_planning_subtree/find_and_hunt_target/mushroom_food
+	target_key = BB_LOW_PRIORITY_HUNTING_TARGET
+	hunting_behavior = /datum/ai_behavior/hunt_target/unarmed_attack_target/mushroom_food
+	hunt_targets = list(/obj/item/food/grown/mushroom)
+	hunt_range = 6
+
+
+/datum/ai_behavior/hunt_target/unarmed_attack_target/mushroom_food
+	hunt_cooldown = 15 SECONDS
+	always_reset_target = TRUE
+
+/mob/living/basic/mushroom/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	. = ..()
+	if(!.)
+		return
+
+	if(!proximity_flag)
+		return
+
+	if(istype(attack_target, /obj/item/food/grown/mushroom))
+		recover(attack_target)
+		return TRUE
+
+/mob/living/basic/mushroom/melee_attack(atom/target, list/modifiers)
+	. = ..()
+
+	if(!.)
+		return
+
+	if(!istype(target, /mob/living/basic/mushroom))
+		return
+	var/mob/living/basic/mushroom/victim = target
+	if(victim.stat != DEAD)
+		return
+	if(victim.faint_ticker < 2)
+		victim.faint_ticker++
+		src.visible_message(span_notice("[src] chews a bit on [victim]."))
+		return
+
+	consume_mushroom(victim)
```

Comment:
```suggestion
	if(victim.faint_ticker >= 3)
		consume_mushroom(victim)
		return

	victim.faint_ticker++
	visible_message(span_notice("[src] chews a bit on [victim]."))
```

`src` is redundant here iirc, and doing it like this (an early return) is both more clear that *3* "faint ticks" is what triggers the consumption and cleaner as far as implementation